detection-rules/credential_phishing_suspicious_file_share_link_or_html.yml

```diff
@@ -0,0 +1,62 @@
+name: "Suspicious file sharing via link or HTML attachment"
+description: "This rule detects messages with verbiage around sharing a file, this has been observed abusing links and HTML documents. Links are further analyzed for credential phishing or suspicious tld indicators. Requires a new or unsolicited sender."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and length(body.current_thread.text) < 800
+  and regex.icontains(body.current_thread.text,
+                      '(shared.{0,30}with you|View Document)'
+  )
+  and (
+    any(body.links,
+        (
+          beta.linkanalysis(.).credphish.contains_login
+          or beta.linkanalysis(.).credphish.contains_captcha
+        )
+  
+        // or linkanalysis concludes phishing of medium to high confidence
+        or any([beta.linkanalysis(.)],
+               (
+                 .credphish.disposition == "phishing"
+                 and .credphish.brand.confidence in ("medium", "high")
+               )
+               or any(file.explode(.screenshot),
+                      strings.contains(.scan.ocr.raw, "link has been removed")
+               )
+        )
+        or .href_url.domain.tld in $suspicious_tlds
+    )
+    or (
+      length(attachments) < 10
+      and any(attachments,
+              (
+                .file_extension in~ ("html", "htm", "shtml", "dhtml")
+                or .file_extension in~ $file_extensions_common_archives
+                or .file_type == "html"
+              )
+      )
+    )
+  )
+  and (
+    (
+      profile.by_sender().prevalence in ("new", "outlier")
+      and not profile.by_sender().solicited
+    )
+  )
+
+
```

Comment:
high_trust_senders here too?

Reply:
high trust senders actually breaks a bunch of the uncovered results, from cloud shares like sharepoint or google
This will flag on sharepoint links that have been deleted too. 

Depending on test-results it might have to be added :)

Reply:
ahh good point... we could allow those specific ones we've seen if having it otherwise provides value